Sources_v3/Controllers/ChannelController/ChannelController.swift

```diff
@@ -266,15 +269,19 @@ public class ChannelControllerGeneric<ExtraData: ExtraDataTypes>: Controller, De
             self.callback { completion?(error) }
         }
         
-        setupEventObservers()
+        if isChannelAlreadyCreated {
+            setupEventObservers()
+        }
     }
 
     // After setting new channelQuery we need to reset current DB observers to observe new channel.
     // Completion will be called if observers will fail to fetch data for the new channel.
     private func set(channelQuery: ChannelQuery<ExtraData>, completion: ((_ error: Error?) -> Void)?) {
-        guard self.channelQuery.cid != channelQuery.cid else { return }
-        self.channelQuery = channelQuery
+        if self.channelQuery.cid != channelQuery.cid {
+            self.channelQuery = channelQuery
+        }
         startDatabaseObservers(completion)
+        setupEventObservers()
```

Comment:
The functionality changed here a bit. Before, we guarded for `cid != channelQuery.cid` and did nothing if it was the same. Now we always start database observers and event observers. Is this intentional?

Reply:
Yes, this is intentional.
If `isChannelAlreadyCreated` is false, we don't setup observers in `startUpdating`. In that case, we'll start observers in `set(query:)` but in this case, if the created channel has the same cid (so if it isn't a DM channel), we won't be setting it here. So, we should still setup observers even if the cid matches.